test/tests/parallel/set3/mta/smtp_delivery.bats

```diff
@@ -227,23 +222,20 @@ function _successful() {
 }
 
 @test "rejects mail to unknown user" {
-  _run_in_container grep '<nouser@localhost.localdomain>: Recipient address rejected: User unknown in virtual mailbox table' /var/log/mail/mail.log
-  assert_success
+  _service_log_should_contain_string 'mail' '<nouser@localhost.localdomain>: Recipient address rejected: User unknown in virtual mailbox table'
   _should_output_number_of_lines 1
 }
 
 @test "redirects mail to external aliases" {
-  _run_in_container_bash "grep 'Passed CLEAN {RelayedInbound}' /var/log/mail/mail.log | grep -- '-> <external1@otherdomain.tld>'"
-  assert_success
-  assert_output --partial '<user@external.tld> -> <external1@otherdomain.tld>'
+  _service_log_should_contain_string 'mail' 'Passed CLEAN {RelayedInbound}'
+  run bash -c "grep '<user@external.tld> -> <external1@otherdomain.tld>' <<< '${output}'"
```

Comment:
This is not viable like similar adjustments you made?

```suggestion
  assert_output --partial '<user@external.tld> -> <external1@otherdomain.tld>'
```

Reply:
Related review comment: https://github.com/docker-mailserver/docker-mailserver/pull/3786#discussion_r1460360285

---

Original:

```bash
# `_should_output_number_of_lines 2` was compatible with these two lines:
_run_in_container_bash "grep 'Passed CLEAN {RelayedInbound}' /var/log/mail/mail.log | grep -- '-> <external1@otherdomain.tld>'"
assert_output --partial '<user@external.tld> -> <external1@otherdomain.tld>'
```

Replacement:

```bash
# assert_output passes but would cause `_should_output_number_of_lines 2`  to fail (13 lines vs 2),
# hence workaround with run bash grep instead of assert_output
_service_log_should_contain_string 'mail' 'Passed CLEAN {RelayedInbound}'
run bash -c "grep '<user@external.tld> -> <external1@otherdomain.tld>' <<< '${output}'"
```

I still don't follow why there is now 13 lines returned from the log method instead of the expected two..

**EDIT:** Ah.. Originally there was a 2nd grep to reduce the output from 13 to 2. The 2nd grep was to filter on the target value `external1@otherdomain.tld`, and use the `assert_output` for the additional context, so that if this test fails we'd get context on the mismatch for that left-side value of the assert - which would not be present by just relying on grep alone.

Reply:
```suggestion
  # Filter to the specific subset of logs retrieved that we're interested in:
  run bash -c "grep -- '-> <external1@otherdomain.tld>' <<< '${output}'"
  # Now assert on the full expected content, if the left-side is a mismatch we'll get the value found via failure context
  assert_output --partial '<user@external.tld> -> <external1@otherdomain.tld>'
```

---

I believe this was done based on the commented out assertion below related to PostSRSd (if it were enabled):

```
assert_output --partial 'external.tld=user@example.test> -> <external1@otherdomain.tld>'
```

Where before it was not obvious why the test failed to match, and without that in the failure context (no results from grep instead), you'd have to do more work to reproduce the log output just to see the failure was due to hat difference in value.